ADAL/src/request/ADAuthenticationRequest+AcquireToken.m

```diff
@@ -227,11 +229,18 @@ - (void)requestToken:(ADAuthenticationCallback)completionBlock
         //The cache lookup and refresh token attempt have been unsuccessful,
         //so credentials are needed to get an access token, but the developer, requested
         //no UI to be shown:
-        ADAuthenticationError* error =
+        ADAuthenticationError* error = _underlyingError ?
```

Comment:
Move the dictionary creation into a separate variable and just pass that into the ADAuthenticationError initializer